packages/neo-one-client/src/user/keystore/LedgerKeyStore.ts

```diff
@@ -0,0 +1,232 @@
+import { AddressString, common, crypto } from '@neo-one/client-core';
+import { Monitor } from '@neo-one/monitor';
+import { utils } from '@neo-one/utils';
+import _ from 'lodash';
+import { BehaviorSubject, Observable } from 'rxjs';
+import { distinctUntilChanged, map } from 'rxjs/operators';
+import { UnknownAccountError } from '../../errors';
+import { publicKeyToAddress } from '../../helpers';
+import { Account, NetworkType, UserAccount, UserAccountID, Witness } from '../../types';
+import { LedgerHandler } from './LedgerHandler';
+
+export interface Ledger {
+  readonly accKey: number;
+  readonly account: UserAccount;
+}
+
+export type Ledgers = { readonly [Network in string]?: { readonly [Address in string]?: Ledger } };
+
+export interface LedgerProvider {
```

Comment:
Actually, you need to take in a provider that knows how to query the right network - this should have one function
```
  readonly getAccount: (network: NetworkType, address: AddressString, monitor?: Monitor) => Promise<Account>;
```

Reply:
You'll want to add the `getAccount` method to `NEOONEProvider` to be passed in to the `LedgerKeyStore`